### PR TITLE
consolidation of linting dependencies, scripts, etc in monorepo

### PR DESCRIPTION
**Pull request by stopfstedt**
- updates all affected dependencies while at it
- adds YAML linting to the mix

this went way out of control, apologies for the mess.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,148 +1,129 @@
 version: 2
```
Comment (jrjohnson) [inline .github/dependabot.yml:1]: I discovered that Dependabot will actually handle all the packages in the monorepo so we don't need to specify them individually. See https://github.com/ilios/frontend/blob/master/.github/dependabot.yml

File: package.json [hunk 2]
```diff
@@ -4,16 +4,38 @@
...
+    "concurrently": "^8.2.2",
+    "ember-template-lint": "^5.13.0",
+    "eslint": "^8.56.0",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-plugin-ember": "^12.0.2",
+    "eslint-plugin-n": "^16.6.2",
+    "eslint-plugin-prettier": "^5.1.3",
+    "eslint-plugin-qunit": "^8.1.1",
+    "eslint-plugin-yml": "^1.12.2",
+    "pre-commit": "^1.2.2",
```
Comment (jrjohnson) [inline package.json:32]: Pre-commit doesn't actually work with `pnpm` this can go along with the `pre-commit` lint specifier.

**Review by jrjohnson — CHANGES_REQUESTED** [review]
Looks good for now. The only thing I'm worried about is we might go back and re-place all of this where it is now for better peer dependency resolution. I'm playing with that in frontend. I'm OK to merge this here with a few changes and then we can go back if needed.

File: .github/dependabot.yml (hunk 1 above)
Comment (stopfstedt) [reply .github/dependabot.yml:1]: consolidated listed dependencies and removed package-specific configuration. 

:white_check_mark:

File: package.json (hunk 2 above)
Comment (stopfstedt) [reply package.json]: removed.

:white_check_mark:

**Comment by stopfstedt** [comment]
> Looks good for now. The only thing I'm worried about is we might go back and re-place all of this where it is now for better peer dependency resolution. I'm playing with that in frontend. I'm OK to merge this here with a few changes and then we can go back if needed.

i'd prefer if we merged this in, it'll give us some cleanup, updated dependencies, and YAML linting. 

will make another pass over this for further realignment with frontend's monorepo layout once that's done.

**Pull request merged**